packages/thirdweb/src/wallets/in-app/native/auth.ts

```diff
@@ -0,0 +1,362 @@
+import type { CognitoUser } from "amazon-cognito-identity-js";
+
+import { Auth } from "aws-amplify";
+import { InAppBrowser } from "react-native-inappbrowser-reborn";
+import type { ThirdwebClient } from "../../../client/client.js";
+import {
+  AuthProvider,
+  type AuthStoredTokenWithCookieReturnType,
+  type OauthOption,
+  RecoveryShareManagement,
+  type SendEmailOtpReturnType,
+} from "../core/authentication/type.js";
+import { ANALYTICS } from "./helpers/analytics.js";
+import {
+  deleteAccount,
+  fetchUserDetails,
+  generateAuthTokenFromCognitoEmailOtp,
+  verifyClientId,
+} from "./helpers/api/fetchers.js";
+import {
+  cognitoEmailSignIn,
+  cognitoEmailSignUp,
+} from "./helpers/auth/cognitoAuth.js";
+import {
+  postAuth,
+  postAuthUserManaged,
+  preAuth,
+} from "./helpers/auth/middleware.js";
+import {
+  DOMAIN_URL_2023,
+  ROUTE_AUTH_ENDPOINT_CALLBACK,
+  ROUTE_AUTH_JWT_CALLBACK,
+  ROUTE_HEADLESS_OAUTH_LOGIN,
+  THIRDWEB_SESSION_NONCE_HEADER,
+} from "./helpers/constants.js";
+import { createErrorMessage } from "./helpers/errors.js";
+import { isDeviceSharePresentForUser } from "./helpers/storage/local.js";
+import { getCognitoUser, setCognitoUser } from "./helpers/storage/state.js";
+import type { VerifiedTokenResponse } from "./helpers/types.js";
+
+const HEADERS = {
+  [THIRDWEB_SESSION_NONCE_HEADER]: ANALYTICS.nonce,
```

Comment:
this... just looks odd? what is this?

Reply:
similar to the other comment this should just import a function that generates the nonce the first time it is used and then re-uses it afterwards, and it should call it *when* the fetch is being made